### PR TITLE
Add non-GUID/string primary keys support for OpenIddictToken

### PR DESCRIPTION
**Pull request by kevinchalet**
Tested with `string`, `long` and `Guid` keys.

@ilmax please review when you have a minute :fries:

### CODE REVIEW

File: src/OpenIddict.Core/Managers/OpenIddictTokenManager.cs [hunk 1]
```diff
@@ -145,19 +145,14 @@ public class OpenIddictTokenManager<TToken, TUser> where TToken : class where TU
         /// <summary>
         /// Creates a new token, defined by a unique identifier and a token type.
         /// </summary>
-        /// <param name="identifier">The unique identifier associated with the token to create.</param>
         /// <param name="type">The token type.</param>
         /// <returns>A <see cref="Task"/> that can be used to monitor the asynchronous operation.</returns>
```
Comment (ilmax) [inline src/OpenIddict.Core/Managers/OpenIddictTokenManager.cs]: Returns message is no more valid

File: src/OpenIddict.Core/Stores/IOpenIddictTokenStore.cs [hunk 2]
```diff
@@ -16,11 +16,10 @@ public interface IOpenIddictTokenStore<TToken> where TToken : class {
         /// <summary>
         /// Creates a new token, defined by a unique identifier and a token type.
         /// </summary>
-        /// <param name="identifier">The unique identifier associated with the token to create.</param>
         /// <param name="type">The token type.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> that can be used to abort the operation.</param>
         /// <returns>A <see cref="Task"/> that can be used to monitor the asynchronous operation.</returns>
```
Comment (ilmax) [inline src/OpenIddict.Core/Stores/IOpenIddictTokenStore.cs]: Returns message is no more valid

File: src/OpenIddict.EntityFramework/Stores/OpenIddictTokenStore.cs [hunk 3]
```diff
@@ -38,24 +38,23 @@ public class OpenIddictTokenStore<TToken, TContext, TKey> : IOpenIddictTokenStor
...
-        public virtual Task CreateAsync(string identifier, string type, CancellationToken cancellationToken) {
+        public virtual async Task<string> CreateAsync(string type, CancellationToken cancellationToken) {
             var converter = TypeDescriptor.GetConverter(typeof(TKey));
 
-            // Ensure that the key type is compatible with string keys.
-            if (!converter.CanConvertFrom(typeof(string))) {
-                return Task.FromResult(0);
+            // Ensure that the key type can be serialized.
+            if (!converter.CanConvertTo(typeof(string))) {
+                throw new InvalidOperationException($"The '{typeof(TKey).Name}' key type is not supported.");
```
Comment (ilmax) [inline src/OpenIddict.EntityFramework/Stores/OpenIddictTokenStore.cs:51]: nice! :tada:

File: src/OpenIddict.Core/Managers/OpenIddictTokenManager.cs (hunk 1 above)
Comment (kevinchalet) [inline src/OpenIddict.Core/Managers/OpenIddictTokenManager.cs]: Well, it is. It is still a task :sweat_smile:

I'll fix it, thanks :smile:

File: src/OpenIddict.EntityFramework/Stores/OpenIddictTokenStore.cs (hunk 3 above)
Comment (kevinchalet) [inline src/OpenIddict.EntityFramework/Stores/OpenIddictTokenStore.cs:51]: Not sure whether `nameof(TKey)` works, I'll give it a try :smile:

File: src/OpenIddict.Core/Stores/IOpenIddictTokenStore.cs [hunk 4]
```diff
@@ -16,11 +16,10 @@ public interface IOpenIddictTokenStore<TToken> where TToken : class {
...
-        Task CreateAsync(string identifier, string type, CancellationToken cancellationToken);
+        Task<string> CreateAsync(string type, CancellationToken cancellationToken);
 
         /// <summary>
         /// Retrieves an token using its unique identifier.
```
Comment (ilmax) [inline src/OpenIddict.Core/Stores/IOpenIddictTokenStore.cs:28]: Retrieves a token

File: src/OpenIddict.EntityFramework/Stores/OpenIddictTokenStore.cs (hunk 3 above)
Comment (kevinchalet) [inline src/OpenIddict.EntityFramework/Stores/OpenIddictTokenStore.cs:51]: Nah, it's not supported. Leaving as-is.

File: src/OpenIddict.Core/Stores/IOpenIddictTokenStore.cs (hunk 4 above)
Comment (kevinchalet) [inline src/OpenIddict.Core/Stores/IOpenIddictTokenStore.cs:28]: I Can Has Cheezburger? :sweat_smile: 

Nice catch, thanks :+1:

**Comment by kevinchalet** [comment]
Updated.

**Comment by ilmax** [comment]
LGTM

**Pull request merged**